File: monopoly.client/src/app/models/player.model.ts
export class PlayerModel implements IPlayerModel {

    public id: string;
    public name: string;
    public color: string;
    public balance: number;
    public property: string[];
    public isArrested: boolean;
    public currentPosition: string;

    constructor(data: IPlayerModel) {
        this.id = data.id;
        this.name = data.name;
        this.color = data.color;
        this.balance = data.balance;
        this.property = data.property;
        this.isArrested = data.isArrested;
        this.currentPosition = data.currentPosition;
    }

}

export interface IPlayerModel {
    id: string;
    name: string;
    color: string;
    balance: number;
    property: string[];
    isArrested: boolean;
    currentPosition: string;
}

File: monopoly.Server/Models/CLient/PlayerModel.cs
namespace monopoly.Server.Models.Client
{
    public class PlayerModel : Entity
    {
        public string Name { get; set; } = string.Empty;
        public float Balance { get; set; } = 1500.0f;
        public List<string> Property { get; set; } = [];
        public bool IsArrested { get; set; } = false;
        public string CurrentPosition { get; set; } = "start";
    }
}
